aviary/core.py

```diff
@@ -397,6 +399,82 @@ def forward(self, *x):
     def num_params(self):
         return sum(p.numel() for p in self.parameters() if p.requires_grad)
 
+    def __repr__(self):
+        name = self._get_name()
+        n_params, n_epochs = self.num_params, self.epoch
+        return f"{name}: {n_params:,} trainable params at {n_epochs:,} epochs"
+
+
+class MessageLayer(nn.Module):
+    """
+    Massage Layers are used to propagate information between nodes in
+    in the stoichiometry graph.
+    """
```

Comment:
Do we want it in core or in segments? Maybe seems like a better fit in segments as CGCNN doesn't use MessageLayer